subgraphs/aggregator/src/index.ts

```diff
@@ -0,0 +1,211 @@
+    userDailyData.transactions = ZERO_BI;
+    userDailyData.user = userID;
+
+    // New aggregator/marketplace user for the day
+    aggregatorDailyData.users = aggregatorDailyData.users.plus(ONE_BI);
+    aggregatorDailyDataByCurrency.users = aggregatorDailyDataByCurrency.users.plus(ONE_BI);
+    marketplaceDailyData.users = marketplaceDailyData.users.plus(ONE_BI);
+    marketplaceDailyDataByCurrency.users = marketplaceDailyDataByCurrency.users.plus(ONE_BI);
+  }
+  userDailyData.transactions = userDailyData.transactions.plus(ONE_BI);
+
+  const userDailyDataByCurrency = getOrInitializeUserDailyDataByCurrency(userDailyData, userByCurrency, dayID);
+  userDailyDataByCurrency.volume = userDailyDataByCurrency.volume.plus(volume);
+
+  // 7. Collection
+  const collectionID = offerToken.toHexString();
+  let collection = Collection.load(collectionID);
+  if (!collection) {
+    collection = new Collection(collectionID);
+    collection.transactions = ZERO_BI;
+
+    // New aggregator/marketplace user
+    aggregator.collections = aggregator.collections.plus(ONE_BI);
+    aggregatorByCurrency.collections = aggregatorByCurrency.collections.plus(ONE_BI);
+    marketplace.collections = marketplace.collections.plus(ONE_BI);
+    marketplaceByCurrency.collections = marketplaceByCurrency.collections.plus(ONE_BI);
+  }
+  collection.transactions = collection.transactions.plus(ONE_BI);
+
+  const collectionByCurrency = getOrInitializeCollectionByCurrency(offerToken, currency);
+  collectionByCurrency.volume = collectionByCurrency.volume.plus(volume);
+  collectionByCurrency.transactions = collectionByCurrency.transactions.plus(ONE_BI);
+
+  // 8. CollectionDailyData
+  const collectionDailyDataID = `${collectionID}-${dayID.toString()}`;
+  let collectionDailyData = CollectionDailyData.load(collectionDailyDataID);
+  if (!collectionDailyData) {
+    collectionDailyData = new CollectionDailyData(collectionDailyDataID);
+    const dayStartTimestamp = dayID.times(ONE_DAY_BI);
+    collectionDailyData.date = dayStartTimestamp;
+    collectionDailyData.transactions = ZERO_BI;
+    collectionDailyData.collection = collectionID;
+
+    // New aggregator/marketplace collection for the day
+    marketplaceDailyData.collections = marketplaceDailyData.collections.plus(ONE_BI);
+    marketplaceDailyDataByCurrency.collections = marketplaceDailyDataByCurrency.collections.plus(ONE_BI);
+    aggregatorDailyData.collections = aggregatorDailyData.collections.plus(ONE_BI);
+    aggregatorDailyDataByCurrency.collections = aggregatorDailyDataByCurrency.collections.plus(ONE_BI);
+  }
+  collectionDailyData.transactions = collectionDailyData.transactions.plus(ONE_BI);
+
+  const collectionDailyDataByCurrency = getOrInitializeCollectionDailyDataByCurrency(collectionByCurrency, dayID);
+  collectionDailyDataByCurrency.volume = collectionDailyDataByCurrency.volume.plus(volume);
+
+  const isBundle = offer.length > 1;
+
+  aggregator.save();
```

Comment:
Can they be saved all at the end?

Reply:
Tests failed if they are done in the end

Reply:
How come?

Reply:
Seems like the transactions are not able to be linked to the `XDailyDataByCurrency`(s) if I save them after saving the transactions